src/onnx/onnx.cpp

```diff
@@ -738,9 +762,21 @@ struct onnx_parser
     static node_map get_nodes(const onnx::GraphProto& graph)
     {
         std::unordered_map<std::string, onnx::NodeProto> result;
+        std::size_t n = 0;
         for(auto&& node : graph.node())
         {
-            result[get_name(node)] = node;
+            if(node.output().empty())
+            {
+                if(node.name().empty())
```

Comment:
If we are assigning the name in this function, does it make sense to modify get_name() above by removing the conditional statement?